### PR TITLE
MWPW-146743 Improve Article Header Performance

### PR DESCRIPTION
**Pull request by meganthecoder**
Made various updates to Article Header in order to reduce LCP on blogs:
* Reduced network requests by pulling code from imports into file
* Delayed placeholder code because it was non-critical
* Only loaded taxonomy if necessary (which it isn't on bacom blog)
* Removed await for build author so requests can be made concurrently

Resolves: [MWPW-146743](https://jira.corp.adobe.com/browse/MWPW-146743)

Results (hlx.live testing):
- Page Speed Index LCP: 0.7s reduction on bacom blog. No change on brand blog.
- DevTools LCP Fast 3G, No CPU throttling: Avg 2.8s reduction on both blogs.
- Incidental reduction in TBT for both blogs

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/methomas/blog-migration/article-header/connecting-external-data-sources-to-adobe-experience-manager-guides
- After: https://perf-refactor-article-header--milo--adobecom.hlx.page/drafts/methomas/blog-migration/article-header/connecting-external-data-sources-to-adobe-experience-manager-guides

Bacom Blog:
- Before: https://main--bacom-blog--adobecom.hlx.live/blog/the-latest/whats-new-at-adobe-summit-2024
- After: https://main--bacom-blog--adobecom.hlx.live/blog/the-latest/whats-new-at-adobe-summit-2024?milolibs=perf-refactor-article-header

Brand Blog:
- Before: https://main--blog--adobecom.hlx.live/en/publish/2024/05/21/lightroom-introduces-power-adobe-firefly-with-generative-remove
- After: https://main--blog--adobecom.hlx.live/en/publish/2024/05/21/lightroom-introduces-power-adobe-firefly-with-generative-remove?milolibs=perf-refactor-article-header

### CODE REVIEW

File: libs/blocks/article-header/article-header.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import { createTag, getMetadata, getConfig } from '../../utils/utils.js';
-import { copyToClipboard } from '../../utils/tools.js';
```
Comment (Brandon32) [inline libs/blocks/article-header/article-header.js:2]: Is `copyToClipboard` in `tools.js` used outside of article header?

File: libs/blocks/article-header/article-header.js [hunk 2]
```diff
@@ -115,37 +144,47 @@ async function buildSharing() {
...
 
-async function validateDate(date) {
+function validateDate(date) {
   const { env } = getConfig();
   if (env?.name === 'prod') return;
   if (date && !/^[0-1]\d{1}-[0-3]\d{1}-[2]\d{3}$/.test(date.textContent.trim())) {
     // match publication date to MM-DD-YYYY format
     date.classList.add('article-date-invalid');
-    date.setAttribute('title', await replaceKey('invalid-date', getConfig()));
+    date.setAttribute('title', 'Invalid Date Format');
```
Comment (Brandon32) [inline libs/blocks/article-header/article-header.js:158]: Looks like this is only used by authoring, could we remove the comment and make the title more descriptive?

**Review by Brandon32 — APPROVED** [review]

File: libs/blocks/article-header/article-header.js (hunk 1 above)
Comment (meganthecoder) [reply libs/blocks/article-header/article-header.js:2]: No, not in milo but I didn't want to delete it in case a consumer is using it.

**Comment by meganthecoder** [comment]
To test invalid date: https://main--milo--adobecom.hlx.page/drafts/methomas/article-header-invalid-date

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**